### PR TITLE
[APR] Fix pool search

### PR DESCRIPTION
**Pull request by luizakp**
This PR:
- Add filter-only pool with voting gauges to APR app with function `filterPoolWithLiveGauge`
  - Therefore, it's not breaking on pool searches anymore
-  Add test for `filterPoolWithLiveGauge`
- Refactor SearchPoolForm to:
  - not submit if a typed pool is not on the gauge json
  - Display a warning: "Only pools with voting gauges displayed"
- Refactor how to get params for `HeaderEndButton.tsx ` and it's redirect

https://www.loom.com/share/f4088580b3ba4b559b8b67ec5138f1ed?sid=4acd794e-349a-41db-b115-146acf81ad64

### CODE REVIEW

File: apps/balancer-tools/src/app/apr/(components)/HeaderEndButton.tsx [hunk 1]
```diff
@@ -42,23 +36,25 @@ export default function HeaderEndButton({
...
+            showPools
+            onlyVotingGauges
           />
         }
       >
-        <div className="flex items-center space-x-2 text-sm font-normal text-slate12 bg-blue3 border border-blue6 p-2 rounded-[4px] cursor-pointer">
+        <div className="flex items-center gap-x-2 text-sm font-normal text-slate12 bg-blue3 border border-blue6 p-2 rounded-[4px] cursor-pointer">
           <MagnifyingGlassIcon width="20" height="20" strokeWidth={1} />
-          <span className="font-medium">Search for specific pool</span>
+          <span className="font-medium pr-1">Open a pool directly</span>
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/(components)/HeaderEndButton.tsx:57]:
```suggestion
          <span className="font-medium pr-1">Go to pool</span>
```

File: apps/balancer-tools/src/components/SearchPoolForm/index.tsx [hunk 2]
```diff
@@ -71,17 +74,19 @@ export function SearchPoolForm({
...
     close?.();
     closeCombobox();
   }
 
-  const filteredPoolList = poolsDataList?.pools.filter((pool) =>
-    filterPoolInput({ poolSearchQuery: poolId, pool }),
-  );
+  const filteredPoolList = poolsDataList?.pools
+    .filter((pool) => filterPoolInput({ poolSearchQuery: poolId, pool }))
+    .filter((pool) => filterPoolWithLiveGauge({ pool, onlyVotingGauges }));
```
Comment (ribeirojose) [inline apps/balancer-tools/src/components/SearchPoolForm/index.tsx:86]: We don't want this filter everywhere. For instance, in the metadata app it's going to hide pools that must be searchable.

File: apps/balancer-tools/src/components/SearchPoolForm/filterPoolWithLiveGauge.test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+describe("filterPoolWithLiveGauge", () => {
+  it("returns true if onlyVotingGauges is false regardless of the pool", () => {
+    const mockPool = { id: "12345" };
+    const result = filterPoolWithLiveGauge({
+      pool: mockPool as Pool,
+      onlyVotingGauges: false,
+    });
+    expect(result).toBeTruthy();
+  });
```
Comment (ribeirojose) [inline apps/balancer-tools/src/components/SearchPoolForm/filterPoolWithLiveGauge.test.ts:14]: not a very useful test IMO 🤷

**Review by ribeirojose — APPROVED** [review]

File: apps/balancer-tools/src/components/SearchPoolForm/index.tsx (hunk 2 above)
Comment (luizakp) [reply apps/balancer-tools/src/components/SearchPoolForm/index.tsx:86]: yup, but this is handled inside filterPoolWithLiveGauge as shown on the video. If `onlyVotingGauges` is false, which is the default value, this filter will have no effect on `poolsDataList?.pools`

**Pull request merged**